source/extensions/filters/network/redis_proxy/config.cc

```diff
@@ -23,16 +23,31 @@ Network::FilterFactoryCb RedisProxyFilterConfigFactory::createFilterFactoryFromP
     Server::Configuration::FactoryContext& context) {
 
   ASSERT(!proto_config.stat_prefix().empty());
-  ASSERT(!proto_config.cluster().empty());
-  ASSERT(proto_config.has_settings());
 
   ProxyFilterConfigSharedPtr filter_config(std::make_shared<ProxyFilterConfig>(
       proto_config, context.scope(), context.drainDecision(), context.runtime()));
-  ConnPool::InstancePtr conn_pool(new ConnPool::InstanceImpl(
-      filter_config->cluster_name_, context.clusterManager(),
-      ConnPool::ClientFactoryImpl::instance_, context.threadLocal(), proto_config.settings()));
-  std::shared_ptr<CommandSplitter::Instance> splitter(new CommandSplitter::InstanceImpl(
-      std::move(conn_pool), context.scope(), filter_config->stat_prefix_, context.timeSource()));
+
+  InstanceFactoryImpl pool_factory(context.clusterManager(), context.threadLocal());
+
+  envoy::config::filter::network::redis_proxy::v2::RedisProxy::RouteConfig route_config(
+      proto_config.route_config());
+
+  // set the catch-all route from the deprecated cluster and settings parameters.
+  if (!route_config.has_catch_all()) {
+    if (route_config.routes_size() == 0) {
+      ASSERT(!proto_config.cluster().empty());
```

Comment:
This config doesn't actually validate this anymore so I think we need an actual error/exception here along with a test that checks it?